src/patternfly/components/Nav/nav.scss

```diff
@@ -1083,6 +1099,10 @@
   &.pf-m-no-title {
     --#{$nav}__section--MarginTop: 0;
   }
+
+  > .#{$nav}__list {
+    --#{$nav}__list--PaddingBottom: 0;
```

Comment:
This is gonna remove the bottom spacing from all lists in a section, removing the spacer from the bottom of the sidebar like this:
<img width="511" alt="Screenshot 2023-05-16 at 7 28 46 PM" src="https://github.com/patternfly/patternfly/assets/35148959/8d3e7e19-f182-4531-bc56-ff60ce13d050">

You could probably do this a couple of ways.

* something like `__section:not(:last-child) { __list--PaddingBottom: 0; }` and target the bottom padding on lists other than the last section.
* just reduce the section + section margin-top to include whatever the nav list padding is
* you might try using a `margin-bottom` on the nav list instead (and heck - `margin-top`, too!)... and [margin-collapse](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Box_Model/Mastering_margin_collapsing) will remove the lists's bottom margin when it touches the section + section margin. Not sure if that would cause any other issues but might be worth trying.